src/adm/filter.rs

```diff
@@ -107,6 +115,104 @@ impl AdmFilter {
         l_sentry::report(&merged_tags, sentry::event_from_error(error));
     }
 
+    /// check to see if the bucket has been modified since the last time we updated.
+    pub async fn requires_update(&self) -> HandlerResult<bool> {
+        // don't update non-bucket versions (for now)
+        if !self.is_cloud() {
+            return Ok(false);
+        }
+        let bucket: url::Url = match self.source.parse() {
```

Comment:
Ah, good call. I had done it as a String mostly to get around the `Default`, and because the source could be a file path and not a URL. I"ll add the extra logic to handle that cleanly.